internal/remote/remote_catalog.go

```diff
@@ -76,22 +76,44 @@ func newRemoteCatalog(kcpClient client.Client, skrContextFactory SkrContextProvi
 	return res
 }
 
-func (c *RemoteCatalog) Sync(
+func (c *RemoteCatalog) SyncModuleCatalog(ctx context.Context, kyma *v1beta2.Kyma) error {
+	moduleReleaseMetas := []v1beta2.ModuleReleaseMeta{}
+	if err := c.GetModuleReleaseMetasToSync(ctx, &moduleReleaseMetas, kyma); err != nil {
+		return err
+	}
+
+	moduleTemplates := []v1beta2.ModuleTemplate{}
+	if err := c.GetModuleTemplatesToSync(ctx, &moduleTemplates, &moduleReleaseMetas); err != nil {
+		return err
+	}
+
+	// https://github.com/kyma-project/lifecycle-manager/issues/2096
+	// Remove this block after the migration to the new ModuleTemplate format is completed.
+	oldModuleTemplate := []v1beta2.ModuleTemplate{}
+	if err := c.GetOldModuleTemplatesToSync(ctx, &oldModuleTemplate, kyma); err != nil {
+		return err
+	}
+	moduleTemplates = append(moduleTemplates, oldModuleTemplate...)
```

Comment:
It is not clear to me if duplicates are possible here. Maybe it is, as the fetching functions already filter the data so that the lists are mutually different, but it's very hard to tell just by looking at the source code.
I understand that this is temporary, Maybe wrapping this in the `filterDuplicates` function - also temporarily - could help?

Reply:
The lists are mutually exclusive. Even if there were duplicate entries, I it wouldn't harm because we would just apply it twice to the SKR (first apply creates it, second apply leaves it unchanged). If you prefer I can still introduce the duplicate check.